ui/hooks/useGasFeeInputs.js

```diff
@@ -221,17 +224,22 @@ export function useGasFeeInputs(
       : null,
   );
   const [gasLimit, setGasLimit] = useState(
-    transaction?.txParams?.gas
-      ? Number(hexToDecimal(transaction.txParams.gas))
-      : 21000,
+    Number(
+      hexToDecimal(
+        transaction?.txParams?.gas ? transaction.txParams.gas : minimumGasLimit,
+      ),
+    ),
   );
+
   const [estimateToUse, setInternalEstimateToUse] = useState(
     transaction
       ? getMatchingEstimateFromGasFees(
           gasFeeEstimates,
           maxFeePerGas,
           maxPriorityFeePerGas,
           gasPrice,
```

Comment:
I think we should only base the match on the maxFeePerGas, maxPriorityFeePerGas and gasPrice. gasLimit is independent of the estimates, and we should show a match with low, medium or high even if the user has changes the gas limit from what it was originally.

Also, keep in mind, when the user clicks the different low, medium and high settings, the gas limit will not change. Gas limit will be the same across all possible gas pricess / fee per gas.